### ISSUE TITLE
Gradle build task failure for `:x-pack:qa:saml-idp-tests:copyIdpFiles`

### ISSUE THREAD

**Issue by n1v0lg** [elastic/elasticsearch#95369]
### CI Link

https://gradle-enterprise.elastic.co/s/yxvvx5uoxrmwk

### Repro line

Can't reproduce locally due to different OSX-related failure

### Does it reproduce?

No

### Applicable branches

main, 8.7

### Failure excerpt

```
Error while evaluating property 'port' of task ':x-pack:qa:saml-idp-tests:copyIdpFiles'.
> Cannot get property 'test.fixtures.shibboleth-idp.tcp.4443' on extra properties extension as it does not exist
```

I see there was a very similar issue a long time ago:

https://github.com/elastic/elasticsearch/issues/40549

Which was related to tests not being skipped correctly in environments where docker wasn't available. However, the code here is quite old, and I'm not spotting any recent changes around the test suite/setup itself. I'm wondering if we made more general changes somewhere around skipping/not skipping tests.

**Comment by mark-vieira** [comment]
I frankly don't know why this happens periodically. The solution here is to ditch docker compose for our test fixtures. It's too much opaque magic that's impossible to debug. We've already done this for the geoip fixture as well as our gcp, s3 and azure fixtures.

**Comment by bpintea** [comment]
One more occurence: https://gradle-enterprise.elastic.co/s/l6qkalz7gdcu6 , 7.17

**Comment by romseygeek** [comment]
And another: https://gradle-enterprise.elastic.co/s/xah2hqdouqkeg/console-log?anchor=16323&page=17

**Comment by kingherc** [comment]
Another one at https://gradle-enterprise.elastic.co/s/hqh276xi4f3au

```
Error while evaluating property 'port' of task ':x-pack:qa:saml-idp-tests:copyIdpFiles'.
> Cannot get property 'test.fixtures.shibboleth-idp.tcp.4443' on extra properties extension as it does not exist
```

**Comment by edsavage** [comment]
Another one at https://gradle-enterprise.elastic.co/s/zscieu3ml37rs

**Comment by edsavage** [comment]
And again: https://gradle-enterprise.elastic.co/s/xazpzjnyn67jm

**Comment by droberts195** [comment]
Another one in https://gradle-enterprise.elastic.co/s/wem4bqe67dug4

**Comment by mark-vieira** [comment]
This has seemingly cropped up again after #100059. I've opened #100206 in hopes that might help.

**Comment by mark-vieira** [comment]
Something is strange that this seems to be consistently this one test fixture, and always the shibboleth service. The only thing that stands out to me here is the `restart: always` bit in the compose file.

@jakelandis do you recall why that container would restart rather than come up correctly on the first attempt. I have a strong suspicion that our docker-compose plugin doesn't take kindly to that behavior.

**Comment by jakelandis** [comment]
> @jakelandis do you recall why that container would restart rather than come up correctly on the first attempt. 

There is a relevant comment here: https://github.com/elastic/elasticsearch/pull/91216#issuecomment-1297724164. Basically it was attempt as a workaround for `Cannot get property 'test.fixtures.shibboleth-idp.tcp.4443' on extra properties extension as it does not exist` which is suspected to be bug in (or at least a misunderstanding of) the docker plugin.

**Comment by mark-vieira** [comment]
> Basically it was attempt as a workaround for `Cannot get property 'test.fixtures.shibboleth-idp.tcp.4443' on extra properties extension as it does not exist` which is suspected to be bug in (or at least a misunderstanding of) the docker plugin.

Hmm, yeah, that's the original problem here but the root cause is still a mystery and it's odd that it's always this container of this test fixture. There's clearly something about it. Why would the container restart? All that config does is say "if this container is stopped, and docker restarts, start it again". That condition should _never_ happen. 🤷

**Comment by mark-vieira** [comment]
Resolved by https://github.com/elastic/elasticsearch/pull/100369
